File: api/models/DeviceOn.js
import mongoose from 'mongoose'

const DeviceOnSchema = new mongoose.Schema(
    {
        mac: {
            type: String,
            required: true,
            unique: false
        },
        ip: {
            type: String,
            required: true,
        },
        date: {
            type: String,
            required: true,
        },
        begin: {
            type: String,
            required: true,
        },
        end: {
            type: String,
            required: true,
        },
    },
);

export default mongoose.model('DeviceOn', DeviceOnSchema);

File: assets/models/deviceon.py
from .db import BasicModel
from utils import get_delta_day
from config import WORKING


class DeviceOn(BasicModel):
    def __init__(self):
        super().__init__(model='deviceons')

    def post(self, request_data):
        data = {'mac': request_data['mac'], 'ip': request_data['ip'], 'end': request_data['time']}
        date = request_data['date']
        if data['end'] <= WORKING['time']['overNight']:
            date = get_delta_day(date, delta=-1)
        data['date'] = date
        previous_data = self.collection.find_one({'mac': data['mac'], 'ip': data['ip'], 'date': data['date']})
        if previous_data is None:
            data['begin'] = request_data['time']
        self.collection.update_one({'mac': data['mac'], 'ip': data['ip'], 'date': data['date']}, {'$set': data}, upsert=True)